pkg/controller/controllerutil/controllerutil.go

```diff
@@ -168,6 +168,36 @@ func RemoveOwnerReference(owner, object metav1.Object, scheme *runtime.Scheme) e
 	return nil
 }
 
+// AppendOwnerReference is a helper method to make sure the given object contains a provided owner reference.
+// This allows you to declare that owner has a dependency on the object without specifying it as a controller.
+// If a reference already exists, it'll be overwritten with the newly provided version.
```

Comment:
This comment is exactly like `SetOwnerReference`. How is this any different besides not using the scheme to get the GVK  and skipping `validateOwner`?

Reply:
I couldn't think of a way to use `Get` in `sigs.ki8s.io/controller-runtime/pkg/client` without know what struct I need in advance (ReplicaSet, Job, etc). If there's a way to do this, then perhaps this entire request is unneeded.

Reply:
This also sidesteps the validation that the function provides when adding an owner reference.  Not sure if want to do that or allow the appending of owner references to any object outside of the scope where the owner would be setting it.  This seems to be very dangerous where the lifecycle is now beholden to an owner reference the owner object didn't explicitly set.

Reply:
I don't follow this second concern. In my experience, it's not at all unusual for a controller to create a resource on behalf of something, and have the resource share the lifecycle of that thing. Is your concern only about resources created by other controllers?